lib/attr_extras/attr_initialize.rb

```diff
@@ -12,22 +12,13 @@ def apply
     names = @names
     block = @block
     validate_arity = method(:validate_arity)
-    set_ivar_from_hash = method(:set_ivar_from_hash)
+    set_ivar_from_args = method(:set_ivar_from_args)
 
     klass.send(:define_method, :initialize) do |*values|
       validate_arity.call(values.length, self.class)
 
       names.zip(values).each do |name_or_names, value|
```

Comment:
I think perhaps this zipping doesn't quite work anymore. I tried adding this spec and it failed. I would guess it's due to mis-zipping but I haven't dug into it:

```
  it "allows mixing optional attributes and hash attributes" do
    klass = Class.new do
      attr_initialize :foo, { :bar => "Bar", :c => "C" }, [ :baz => "Baz", :d => "D" ]
    end

    example = klass.new("Foo", "Rab", :baz => "Zab")
    example.instance_variable_get("@bar").must_equal "Rab"
    example.instance_variable_get("@baz").must_equal "Zab"
    example.instance_variable_get("@c").must_equal "C"
    example.instance_variable_get("@d").must_equal "D"
  end
```